script/uninstall.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env sh
+set -eu
+
+# Uninstalls Zed that was installed using the install.sh script
+# Modeled after install.sh
+
+main() {
+    platform="$(uname -s)"
+    channel="${ZED_CHANNEL:-stable}"
+
+    if [ "$platform" = "Darwin" ]; then
+        platform="macos"
+    elif [ "$platform" = "Linux" ]; then
+        platform="linux"
+    else
+        echo "Unsupported platform $platform"
+        exit 1
+    fi
+
+    "$platform"
+
+    echo "Zed has been uninstalled"
+}
+
+linux() {
+    suffix=""
+    if [ "$channel" != "stable" ]; then
+        suffix="-$channel"
+    fi
+
+    appid=""
+    case "$channel" in
+      stable)
+        appid="dev.zed.Zed"
+        ;;
+      nightly)
+        appid="dev.zed.Zed-Nightly"
+        ;;
+      preview)
+        appid="dev.zed.Zed-Preview"
+        ;;
+      dev)
+        appid="dev.zed.Zed-Dev"
+        ;;
+      *)
+        echo "Unknown release channel: ${channel}. Using stable app ID."
+        appid="dev.zed.Zed"
+        ;;
+    esac
+
+    # Remove the app directory
+    rm -rf "$HOME/.local/zed$suffix.app"
+
+    # Remove the binary symlink
+    rm -f "$HOME/.local/bin/zed"
+
+    # Remove the .desktop file
+    rm -f "$HOME/.local/share/applications/${appid}.desktop"
+}
+
+macos() {
+    app="Zed.app"
+    case "$channel" in
+      nightly)
+        app="Zed Nightly.app"
+        ;;
+      preview)
+        app="Zed Preview.app"
+        ;;
+      dev)
+        app="Zed Dev.app"
+        ;;
+    esac
+
+    # Remove the app bundle
+    if [ -d "/Applications/$app" ]; then
+        rm -rf "/Applications/$app"
+    fi
+
+    # Remove the binary symlink
+    rm -f "$HOME/.local/bin/zed"
+}
```

Comment:
Similarly, there's `~/.config/zed/` on both Linux and macOS, which I'm not certain what to do with, if someone has more than one version installed.

I wonder if we should prompt for deleting that and let the user to decide?

Reply:
Yeah, sure, would the following work:
```bash
printf "Do you want to keep your Zed preferences? [Y/n] "
read -r response
case "$response" in
    [nN]|[nN][oO])
        rm -rf "$HOME/.config/zed"
        echo "Preferences removed."
        ;;
    *)
        echo "Preferences kept."
        ;;
esac
```

Reply:
Come to think of it, maybe we can check if there are other versions, via https://github.com/zed-industries/zed/pull/21213#discussion_r1860559294 , and only remove it if it's the only version in the system, being removed.